### DISCUSSION TITLE
Solver debug run

### DISCUSSION THREAD

**Discussion by veera-visw** [pybamm-team/PyBaMM#3809] [General]
I am working on a battery model based off basic_dfn.py with some modifications to the source terms of the equations. As expected, my first attempt is not working (:D) and I get following error:
**_Could not find consistent states: Could not find acceptable solution: Error in Function::call for 'roots' [Newton] at .../casadi/core/function.cpp:330:
.../casadi/core/rootfinder.cpp:277: rootfinder process failed. Set 'error_on_fail' option to false to ignore this error._**

Just curious if there is a way to run a debug/diagnostic version of the solver with more verbose prints so I can identify which equation is struggling (for example).

Thanks

**Comment by agriyakhetarpal** [comment]
Hi @veera-visw, does something of this sort at the top of your script help you?

```python
import pybamm
pybamm.set_logging_level("DEBUG"). # i.e., DEBUG instead or INFO
```

This will enable more verbose logging for the various events involved with solving the model equations.

If you could share the code snippet that caused the failure or a minimal example for it, that would be helpful for further investigation

**Reply by veera-visw** [reply]
Thanks @agriyakhetarpal  Good to know about the debug mode, but this did not help me with the problem. I think issue was in the model formulation, as I got it to work now after rewriting the model from scratch.

A debug mode that would be helpful (which I might try to do in my code next) is a way to plot results on the run so we can see which variable is exploding (or causing problems).  If there is a possibility to do that already, pelase let me know :)

**Reply by agriyakhetarpal** [reply]
Glad to hear it works now, @veera-visw! I am not sure how to plot results "on the run" as such besides extracting such data from what you receive from the solver, perhaps someone better acquainted with running real-world experiments can help :)
